core/schema_utils.py

```diff
@@ -134,7 +136,18 @@ def normalize_against_schema(
                 prop[SCHEMA_KEY_SCHEMA],
                 global_validators=global_validators
             )
+    elif schema[SCHEMA_KEY_TYPE] == SCHEMA_TYPE_DICT_WITH_VARIABLE_NO_OF_KEYS:
+        assert isinstance(obj, dict), ('Expected dict, received %s' % obj)
+        schema_value_type = schema[SCHEMA_KEY_VALUES]
+        normalized_obj = {}
+        for key, value in obj.items():
+            normalized_obj[key] = normalize_against_schema(
+                value, schema_value_type[SCHEMA_KEY_SCHEMA],
+                global_validators=global_validators
+            )
     elif schema[SCHEMA_KEY_TYPE] == SCHEMA_TYPE_FLOAT:
+        if isinstance(obj, bool):
+            raise Exception('Expected float, received %s' % obj)
```

Comment:
Why are you adding this?

Reply:
Since after adding schema validation the values are normalized for example after validation values of type `Bool(True)` is normalized to `float(1.0)`
https://github.com/oppia/oppia/blob/66ce06abf05ee26fc8613836e93fe8451ee2ba05/core/controllers/skill_mastery_test.py#L345
and so the below part will never run in handlers
https://github.com/oppia/oppia/blob/66ce06abf05ee26fc8613836e93fe8451ee2ba05/core/controllers/skill_mastery.py#L93
so I added a check in schema_utils to avoid such instances.